lockup/README.md

```diff
@@ -1,107 +1,110 @@
-lockup period.
+Once transfers are enabled, all tokens are unlocked, everything is vested, and the termination process is ended (if applicable), the owner can add a full access key to the account.
+This will allow the owner to turn this contract account into a regular account, claim the remaining tokens, and remove the contract or delete the account.
 
-Once all tokens are unlocked, everything is vested, and transfers are enabled, the owner can add a full access key to the
-account. This will allow the owner to turn this contract account into a regular account, claim the remaining tokens, and remove the contract
-or delete the account.
+### Lockup schedule
+
+Lockup is a mechanism of linear unlocking of tokens that could not be terminated.
+It is described by the following fields:
+- `lockup_timestamp` - The moment when the tokens become unlock (start of the contract).
+- `lockup_duration` - Alternative way to set the moment when the tokens become unlock.
+  The duration from the moment the transfers are enabled to the start of the contract.
+- `release_duration` - Length of the contract. `finish_timestamp = lockup_timestamp + release_duration`
 
 ### Vesting schedule
 
-The contract can contain a vesting schedule and serve as a vesting agreement between the NEAR Foundation (Foundation) and an employee (owner of contract).
-Vesting schedule is described by three timestamps in nanoseconds:
+The contract can contain a vesting schedule and serve as a vesting agreement between the Contract Creator and an employee (owner of the contract).
+
+Contract Creator is either the NEAR Foundation, or it could be a third-party account.
+It is set at the moment of initializing the contract by the `foundation_account_id` field (naming could be confusing).
+It's also possible to leave this field blank, which means that the vesting process could not be terminated.
+
+A vesting schedule is described by three timestamps in nanoseconds:
 - `start_timestamp` - When the vesting starts. E.g. the start date of employment.
 - `cliff_timestamp` - When the first part of lockup tokens becomes vested.
- The remaining tokens will vest continuously until they are fully vested.
- Example: a 1 year of employment at which moment the 1/4 of tokens become vested.
+  The remaining tokens will vest continuously until they are fully vested.
+  Assume we have a 4-year contract with a 1-year cliff.
+  In the first year, nothing is vested, then 25% is vested, then we have linear vesting till the end of the contract.
 - `end_timestamp` -  When the vesting ends.
 
-In addition to the lockup period that starts from the moment the transfers are enabled, vesting schedule also locks
-all tokens until `cliff_timestamp`.
-Once the `cliff_timestamp` passed, the tokens are vested on a pro rata basis from the `start_timestamp` to the `end_timestamp`.
+Once the `cliff_timestamp` passed, the tokens are vested on a pro-rata basis from the `start_timestamp` to the `end_timestamp`.
```

Comment:
Are you sure that example above is not enough? (5 lines before)
Don't want to overload the doc.

We also have many examples [here](https://github.com/telezhnaya/docs/blob/master/docs/tokens/lockup_vesting_details.md), it will be merged into [our doc](https://docs.near.org/docs/tokens/lockup), I think we don't need additional example here.